library/std/src/io/error.rs

```diff
@@ -178,29 +269,54 @@ pub enum ErrorKind {
     /// read.
     #[stable(feature = "read_exact", since = "1.6.0")]
     UnexpectedEof,
+
+    /// This operation is not supported on this platform.
+    #[stable(feature = "not_supported_error", since = "1.50.0")]
+    NotSupported,
```

Comment:
I don't think this is the right name for `ENOSYS`. It's already challenging, sometimes, to map these names to the underlying errno code. This name, in particular, seems much more likely to mean `EOPNOTSUPP` rather than `ENOSYS`.

Reply:
Thanks for the attention to detail.

Please see #78880 where this is added - this is not my commit.  For obvious reasons this branch needs to be on top of that one...

Reply:
Ah. I didn't catch that this came from another PR.

I've posted a comment there.